source/NPC.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include"Item.hpp"
+
+#include<string>
+#include<vector>
+#include<random>
+#include<memory>
+
+namespace dungeon {
+
+struct TradingItem {
+	std::unique_ptr<Item> item;
+	int price;
+	TradingItem(std::unique_ptr<Item>&&, int);
+};
+
+struct NPC {
+	std::string name;
+	std::mt19937 rndGen;
```

Comment:
IMHO it is better to use a single random generator throughout the whole program to prevent from time based attack.